app/js/components/names.jsx

```diff
@@ -1,265 +1,241 @@
-                                ref="familyname"
-                                name="familyName"
-                                type="text"
-                                value={this.state.familyName}
-                                onChange={this.handleChange}
-                                readOnly={this.state.display === 'view' ? 'readonly' : null}
-                                required />
-                    </div>
-                </div>
-
-                <div className="form-group ">
-                    <label className="control-label col-sm-2"> Created By: </label>
-                    <div className="col-sm-5">
-                        <input  className="form-control"
-                                name="createdby"
-                                type="text"
-                                size="50"
-                                value= {this.state.creator +'  ' + this.state.dateCreated}
-                                disabled />
-                    </div>
-                </div>
-                {this.state.display == "edit" &&
-                <div className="form-group">
-                    <div className="col-sm-2">
-                        <button type="submit"
-                             name="update"
-                             onClick={this.handleSave}
-                             className="btn btn-success form-control">
-                             Save</button>
-                    </div>
-                    <div className="col-sm-2">
-                        <button type="button"
-                             name="cancel"
-                             onClick={this.handleCancel}
-                             className="btn btn-default form-control cancelBtn">
-                             Cancel</button>
-                    </div>
-               </div>
-            }
-
-            </form>
-            </div>
-        );
-    };
+    if(!this.state.familyName) {
+      this.setState({ errorFamilyName: 'Family Name can not be null' })
+    }
+  }
+
+  handleCancel() {
+    this.fetch();
+    const given = this.state.response.preferredName.givenName;
+    const middle = this.state.response.preferredName.middleName;
+    const family = this.state.response.preferredName.familyName;
+    this.setState((prevState) => ({
+      display: 'view',
+      givenName: given === null ? '' : prevState.givenName,
+      middleName: middle === null ? '' : prevState.middleName,
+      familyName: family === null ? '' : prevState.familyName,
+      voided: prevState.response.preferredName.voided,
+      preferred: prevState.preferred,
+      creator: prevState.response.auditInfo.creator.display,
+      dateCreated: prevState.response.auditInfo.dateCreated,
+      errorGivenName: '',
+      errorFamilyName: ''
+    }));
+  };
+
+  render() {
+    let givenErrorClass = '';
+    let familyErrorClass = '';
+    if(this.state.errorGivenName && this.state.display === 'edit') {
```

Comment:
You could deconstruct the state (something like `const { errorGivenName, display } = this.state;` so that you can do something like `if (errorGivenName && display) `. This will reduce the column width and save you some stress maybe :). This implementation is cool though. 👍

Reply:
Thanks, i think this one is a judgement call. Since this deconstruction will not save me any LOC seeing as the if statement is  within 80 chars, i will maintain it. Thanks for the review though.